python_modules/dagster/dagster/_core/definitions/partition_mapping.py

```diff
@@ -201,7 +201,9 @@ def get_downstream_partitions_for_partitions(
         current_time: Optional[datetime] = None,
         dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
     ) -> PartitionsSubset:
-        raise NotImplementedError()
+        return downstream_partitions_def.subset_with_all_partitions(
```

Comment:
Can we check if the upstream subset is empty? If it's empty, we should also return an empty subset (since all the downstream asset's partitions depend on any partition in the upstream asset)

Reply:
okay, pushed https://github.com/dagster-io/dagster/pull/19375/commits/a89f57f73fe755e9c841b406d041437b2c11cfde to return `empty_subset()` if the length of the upstream subset is zero.